datalad/support/protocol.py

```diff
@@ -0,0 +1,147 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+""" Protocolling  command calls.
+"""
+
+from os import linesep
+
+
+class ProtocolInterface(object):
+    """Interface class for protocols used by the Runner.
+    """
+
+    def __init__(self):
+        pass
```

Comment:
there is also an https://docs.python.org/2/library/abc.html  abc module for writing proper abstract classes etc... I am yet to start using it myself